### PR TITLE
Adding a isComplete function #12

### PR DESCRIPTION
**Pull request by ppivanov**
I have no idea how much of this works as my printer's been dead for a while, but whoever needs it can test it out and share some feedback or update the branch accordingly.

Updating `print` to return the job id based on what @artiebits commented on  #12.
Adding `isPrintComplete` to query the queued jobs and will return false if it finds a row matching the job id (I'm very much assuming it should disappear when it completes).

Whoever utilizes the package can add custom polling / delays as calling `isPrintComplete` just once immediately after `print` will likely always return false.

Docs will need to be updated if you wanna go through with it

### CODE REVIEW

File: src/print/print.ts
Comment (artiebits) [inline src/print/print.ts:28]: Hi @ppivanov, thank you very much for your contribution. I would like to see your work in master but I have a few remarks to consider.

Currently, the print function returns useful information in `{stdout, stderr}`. Would it be better to leave it as is and simply provide a new isPrintComplete function where the user can pass a print request id? For now, I think we can leave it to the user to parse the `'request ID is myDummyPrinter-4 (1 file(s))'` message. Additionally, will your `getJobId` function work if the printer name is `my-dummy-printer-4`?

I also think that the `isPrintComplete` function should accept the request id instead of the job id. This is because there might be many printers in a logistics department, for example, and it's possible that they might have the same job id at some point.

Comment (ppivanov) [reply src/print/print.ts:28]: > will your getJobId function work if the printer name is my-dummy-printer-4

I thought I took such cases into account, but it seems I only it was only a thought. It most definitely will not work as is!

We can most certainly change `getJobId` to return the request id instead.

File: src/utils/parse-response.spec.ts
Comment (ppivanov) [inline src/utils/parse-response.spec.ts:34]: All tests in here are now stale and there's some additional stuff to be tested

File: src/print/print.ts
Comment (ppivanov) [inline src/print/print.ts:24]: stale

File: src/index.ts
Comment (artiebits) [inline src/index.ts:4]: could you please change it to `export { default as isPrintComplete }` to be able to import it in a project like this `import { isPrintComplete } from "unix-print"`

File: src/print/print.ts
Comment (artiebits) [inline src/print/print.ts:8]: why have you removed the `Promise<void>`?

File: src/utils/parse-response.ts
Comment (artiebits) [inline src/utils/parse-response.ts:7]: could you please delete the comments? all things that are expected can be tested with unit tests

File: src/utils/parse-response.ts
Comment (artiebits) [inline src/utils/parse-response.ts:29]: please delete the comment

**Review by artiebits — CHANGES_REQUESTED** [review]
Hi @ppivanov, thank you very much for the contribution. i would love to see this functionality in master, and i have just a few small comments.

File: src/print/print.ts
Comment (ppivanov) [reply src/print/print.ts:8]: It actually returns the output of exec so it's misleading. Can put it back but I think type errors will start popping up if you try to use the new function

File: src/index.ts
Comment (ppivanov) [reply src/index.ts:4]: You can import it as `import { isPrintComplete } from "unix-print"` as it currently is but I'll update it to match the remaining exports

File: src/print/print.ts
Comment (ppivanov) [reply src/print/print.ts:8]: Adding an explicit return type

**Comment by artiebits** [comment]
hey, I wanted to check if this work is finished. It still says "DO NOT MERGE"

**Comment by ppivanov** [comment]
Hey! Still don't have the hardware to test this functionally. If anyone has gotten a chance to make sure it works I'm all for it.

**Comment by ppivanov** [comment]
A new printer is on its way... Will be able to test this soon 🤞

**Comment by ppivanov** [comment]
this should be fine now. would have loved someone else giving it a spin in case something obvious was missed

**Comment by artiebits** [comment]
hi @ppivanov thanks for your work! I have published a beta version `unix-print@1.3.0-beta.1` to test it. let me know if it works for you, i am testing it too.

**Comment by shubhamgajra-memechat** [comment]
Why is a loop needed to check for print completion?

**Comment by ppivanov** [comment]
> Why is a loop needed to check for print completion? 

If the job is still on the queue the first time you check (e.g. immediately after it was submitted) you may want to check again after a minute or two have passed. There are probably better polling techniques that won't block the thread.

**Comment by shubhamgajra-memechat** [comment]
Any update on this @artiebits  ?

**Comment by artiebits** [comment]
hi @ppivanov, sorry for such a big delay in reviewing your PR. it looks good to me, and the only comment I have is that the example in readme doesn't look valid to me because it will execute the check 10 times without any delay. 
As an option, keep it consistent with other examples in the readme and do:
```
isPrintComplete(printJob).then(console.log)
```

or something like this, which will stop checking If the print job is complete in the first iteration of the loop:
```
async function waitForPrintCompletion(printJob) {
  while (!await isPrintComplete(printJob)) {
    // Wait a bit before checking again (to avoid constant checks)
    await new Promise(resolve => setTimeout(resolve, 1000)); // Wait for 1 second
  }
  console.log('Job complete');
}

await waitForPrintCompletion(printJob);
```

let's keep it simple for now!

**Comment by shubhmg** [comment]
Can a new version be released as soon as this is merged, thanks, great work @artiebits @ppivanov! Really useful library

**Review by artiebits — APPROVED** [review]

**Comment by artiebits** [comment]
@ppivanov thank you for your contribution! I will release a new version today

**Pull request merged**
